packages/tenant-consumer/src/tenantConsumerService.ts

```diff
@@ -1,19 +1,54 @@
 import { match } from "ts-pattern";
-import { logger } from "pagopa-interop-commons";
+import {
+  ReadModelRepository,
+  consumerConfig,
+  logger,
+} from "pagopa-interop-commons";
 import { EventEnvelope } from "./model/models.js";
+import { fromTenantV1 } from "./model/converter.js";
+
+const { tenants } = ReadModelRepository.init(consumerConfig());
 
 export async function handleMessage(message: EventEnvelope): Promise<void> {
   logger.info(message);
   await match(message)
-    .with({ type: "TenantCreated" }, async (_msg) => {
-      logger.info("TODO");
+    .with({ type: "TenantCreated" }, async (msg) => {
+      await tenants.updateOne(
+        {
+          "data.id": msg.stream_id,
+        },
+        {
+          $setOnInsert: {
+            data: msg.data.tenant ? fromTenantV1(msg.data.tenant) : undefined,
+            metadata: {
+              version: msg.version,
+            },
+          },
+        },
+        { upsert: true }
+      );
     })
     .with({ type: "TenantDeleted" }, async (_msg) => {
       logger.info("TODO");
     })
-    .with({ type: "TenantUpdated" }, async (_msg) => {
-      logger.info("TODO");
-    })
+    .with(
+      { type: "TenantUpdated" },
```

Comment:
Event already present in these PRs
- https://github.com/pagopa/interop-be-monorepo/pull/132
- https://github.com/pagopa/interop-be-monorepo/pull/123

Reply:
Yes, we are aware of it. Sorry for the overhead in review.